packages/devtools_app/lib/src/screens/vm_developer/vm_class_screen_controller.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../primitives/auto_dispose.dart';
+import '../../service/vm_service_wrapper.dart';
+import '../../shared/globals.dart';
+import '../debugger/debugger_model.dart';
+
+/// Auto DisposableController for fetching and displaying data on the class screen.
+class ClassScreenController extends DisposableController
+    with AutoDisposeControllerMixin {
+  ClassScreenController(ClassRef? classRef) {
+    _classRef = classRef;
+    refresh();
+  }
+
+  VmServiceWrapper get _service => serviceManager.service!;
+
+  IsolateRef? _isolate;
+  ClassRef? _classRef;
+
+  Class? get clazz => _clazz;
+  Class? _clazz;
+
+  InstanceSet? get instances => _instances;
+  InstanceSet? _instances;
+
+  String? get scriptUri => _scriptUri;
+  String? _scriptUri;
+
+  SourcePosition? get pos => _pos;
+  SourcePosition? _pos;
+
+  ValueListenable<bool> get refreshing => _refreshing;
+  final _refreshing = ValueNotifier<bool>(false);
```

Comment:
I don't think we are using this in the right way. Are you trying to use this ValueNotifier as a way to force the UI to rebuild when `refresh` is clicked? AFAICT we never actually do anything with the value of this notifier.

Reply:
I do think we should keep this listenable, but we should be using it to determine whether or not we display a spinner. We could update `refresh()` below to do something like this:

```dart
Future<void> refresh() async {
  _refreshing.value = true;
  final isolateRef = ...
  // Refresh logic here.
  _refreshing.value = false;
}
```

And then use the `ValueListenableBuilder<bool>` to display the spinner when `_refreshing.value = true`.